src/EPG.cpp

```diff
@@ -132,13 +129,65 @@ PVR_ERROR EPG::GetEPGForChannel(int channelUid, time_t start, time_t end, kodi::
         }
 
       }
-      broadcast.SetSeriesNumber(XMLUtils::GetIntValue(pListingNode, "season", EPG_TAG_INVALID_SERIES_EPISODE));
-      broadcast.SetEpisodeNumber(XMLUtils::GetIntValue(pListingNode, "episode", EPG_TAG_INVALID_SERIES_EPISODE));
+
+      int season{ EPG_TAG_INVALID_SERIES_EPISODE };
+      int episode{EPG_TAG_INVALID_SERIES_EPISODE};
+      XMLUtils::GetInt(pListingNode, "season", season);
+      XMLUtils::GetInt(pListingNode, "episode", episode);
+      broadcast.SetEpisodeNumber(episode);
       broadcast.SetEpisodePartNumber(EPG_TAG_INVALID_SERIES_EPISODE);
+      // Backend could send epidode only as S00 and parts are not support
+      if (season <= 0 || episode == EPG_TAG_INVALID_SERIES_EPISODE)
+      {        
+        std::regex base_regex("^.*\\([eE][pP](\\d+)(?:/?(\\d+))?\\)");
+        std::smatch base_match;
+        if (std::regex_search(description, base_match, base_regex))
+        {
+          broadcast.SetEpisodeNumber(std::atoi(base_match[1].str().c_str()));
+          if (base_match[2].matched)
+            broadcast.SetEpisodePartNumber(std::atoi(base_match[2].str().c_str()));
+        }
+        else if (std::regex_search(description, base_match, std::regex("^([1-9]\\d*)/([1-9]\\d*)\\.")))
+        {
+          broadcast.SetEpisodeNumber(std::atoi(base_match[1].str().c_str()));
+          broadcast.SetEpisodePartNumber(std::atoi(base_match[2].str().c_str()));
+        }
+      }
+      if (season != EPG_TAG_INVALID_SERIES_EPISODE)
+      {
+        // clear out NextPVR formatted data, Kodi supports S/E display
+        if (subtitle == kodi::tools::StringUtils::Format("S%02dE%02d", season, episode))
+        {
+          subtitle.clear();
+        }
+        if (season == 0)
+          season = EPG_TAG_INVALID_SERIES_EPISODE;
+      }
+      broadcast.SetSeriesNumber(season);
+      broadcast.SetEpisodeName(subtitle);
+
+      int year{-1};
+      if (XMLUtils::GetInt(pListingNode, "year", year))
+      {
+        broadcast.SetYear(year);
+      }
 
       std::string original;
-      XMLUtils::GetString(pListingNode, "original", original);
-      broadcast.SetFirstAired(original);
+      if (XMLUtils::GetString(pListingNode, "original", original))
+      {
+        if (broadcast.GetGenreType() == 16 && broadcast.GetGenreSubType() == 0 && year == -1 && original.length() > 4)
```

Comment:
16? Magic number anti pattern! Maybe introduce a constant so that one can see immediately what this value stands for. Same for the magic 4 and 0 in this line.

Reply:
OK use system defines and comment on the 4 instead of constant